packages/react-performance/src/context.ts

```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import {Performance} from '@shopify/performance';
 
-export const PerformanceContext = React.createContext<Performance | undefined>(
-  undefined,
+import {isServer} from './utilities';
+
+export const PerformanceContext = React.createContext<Performance | null>(
+  isServer() ? null : new Performance(),
```

Comment:
personally I prefer not having utilities for this kind of thing, because it's something we may want to some day optimize in build, and it's very hard to do that when programatic APIs wrap the statically-analyzable bits.